tests/components/ai_task/test_task.py

```diff
@@ -0,0 +1,93 @@
+"""Test tasks for the AI Task integration."""
+
+from freezegun import freeze_time
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.ai_task import DATA_PREFERENCES, async_generate_text
+from homeassistant.components.conversation import async_get_chat_log
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import chat_session
+
+from .conftest import TEST_ENTITY_ID
+
+
+async def test_run_task_preferred_entity(
+    hass: HomeAssistant,
+    init_components: None,
+) -> None:
+    """Test running a task with an unknown entity."""
+    preferences = hass.data[DATA_PREFERENCES]
+
+    with pytest.raises(
+        ValueError, match="No entity_id provided and no preferred entity set"
+    ):
+        await async_generate_text(
+            hass,
+            task_name="Test Task",
+            instructions="Test prompt",
+        )
+
+    preferences.async_set_preferences(gen_text_entity_id="ai_task.unknown")
```

Comment:
These tests that muck with the preferences object feel like they're working behind the public APIs which is normally preferred in tests. I think this type of test should use the websocket API?

(Not sure how i feel about the tests for writing the preference storage, which should theorietically be similar, but feels worth testing at a low level still)

Reply:
for me feels similar to how we interact with device registry, also within Python.

Reply:
I think the part that is different is using hass.data to get it, so maybe giving an API to get the prefs would make it more like device registry. If its not part of the API then its different from the device registry.